src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -77,6 +77,47 @@
 public final class Matches {
   private Matches() {}
 
+  /**
+   * Returns a match whose condition is always satisfied.
+   *
+   * @return A match; never {@code null}.
+   */
+  public static <T> Match<T> always() {
+    return Match.of(it -> true);
+  }
+
+  /**
+   * Returns a match whose condition is never satisfied.
+   *
+   * @return A match; never {@code null}.
+   */
+  public static <T> Match<T> never() {
+    return Match.of(it -> false);
+  }
+
+  /**
+   * Returns the number of matches found.
+   */
+  public static <T> int countMatches(final Collection<T> collection, final Match<T> match) {
+    return (int) collection.stream().filter(match::match).count();
+  }
+
+  /**
+   * Returns the elements of the collection that match.
+   */
+  public static <T> List<T> getMatches(final Collection<T> collection, final Match<T> match) {
+    return collection.stream().filter(match::match).collect(Collectors.toList());
+  }
+
+  /**
+   * Only returns the first n matches.
+   * If n matches cannot be found will return all matches that
+   * can be found.
+   */
+  public static <T> List<T> getNMatches(final Collection<T> collection, final int max, final Match<T> match) {
+    return collection.stream().filter(match::match).limit(max).collect(Collectors.toList());
+  }
+
```

Comment:
The only minor concern I have here is that `Matches` is essentially a collection of game logic predicates, whereas these methods are utilities that operate on generic collections _using_ a predicate.  It seems like those are separate concerns and should be in different classes.  In the future, when `Match<T>` is replaced with `Predicate<T>`, I could imagine these methods being viewed as `Collection` helpers that have nothing to do with TripleA game logic and be located in some kind of `Collection`-specific utility class:

```java
package games.strategy.util;

public class CollectionUtils {
  public static <T> long countMatches(
      final Collection<T> collection,
      final Predicate<? super T> predicate) {
    // ...
  }

  public static <T> Collection<T> filter(
      final Collection<T> collection,
      final Predicate<? super T> predicate) {
    // ...
  }

  public static <T> Collection<T> filter(
      final Collection<T> collection,
      final Predicate<? super T> predicate,
      final long maxSize) {
    // ...
  }
}
```

Reply:
I think this is really a minor concern :D
We can always change that in the future, to have seperate classes for different purposes.

What would be worth thinking about, is passing Streams instead of collections, as streams are lazy data collections e.g. the data is only computed if it's actually needed, which would save us many unnecessary iterations over collections and possibly a good bit if memory and therefore boost performance.